src/components/Canvas.tsx

```diff
@@ -278,7 +291,32 @@ const Canvas: React.FC<CanvasProps> = ({
     return () => {
       window.removeEventListener("resize", handleResize);
     };
-  }, [channels]);
+  }, [channels, updateChartColors]);
+
+  useEffect(() => {
+    const updateChannels = () => {
+      channels.forEach((_, index) => {
+        if (chartRef.current[index]) {
+          const chart = chartRef.current[index];
+          chart.options.maxValue = shouldAutoScale(selectedBits)
+            ? undefined
+            : getMaxValue(selectedBits);
+          chart.options.minValue = 0;
+
+          const series = seriesRef.current[index];
+          if (series) {
+            chart.removeTimeSeries(series);
```

Comment:
This may be occurring issue, also I am unable to understand role of the function

Reply:
We can't modify the TimeSeries options directly, but we can update the options for the series by removing it and re-adding it with the updated options.